### PR TITLE
search UUID and description fixes #1330

### LINKED ISSUE #1330: Search for video UUID should return that video
**What happened?**

Pasted a video UUID into search (verified that the UUID worked when constructing a URL with it), no results returned.

**What do you expect to happen instead?**

Expected a search for any exact identifier for that video to return the video.

**Steps to reproduce:**

1. Paste UUID into search

### PR DESCRIPTION
**Pull request by McFlat**
Haven't tested it but it should work, will have to test it sometime today once I get some videos imported locally.

fixes #1330 

- make video search to search uuid instead of only full uri to the video, include description field in search
- attributeInclude uuid, description just in case

### CODE REVIEW

File: server/models/video/video.ts [hunk 1]
```diff
@@ -1082,8 +1082,11 @@ export class VideoModel extends Model<VideoModel> {
               '(' +
               'SELECT "video"."id" FROM "video" ' +
               'WHERE ' +
+              'lower(immutable_unaccent("video"."uuid")) % lower(immutable_unaccent(' + escapedSearch + ')) OR ' +
```
Comment (Chocobozzz) [inline server/models/video/video.ts]: We want an exact search for the uuid, we don't need the similarity

File: server/models/video/video.ts [hunk 2]
```diff
@@ -1082,8 +1082,11 @@ export class VideoModel extends Model<VideoModel> {
...
               'lower(immutable_unaccent("video"."name")) % lower(immutable_unaccent(' + escapedSearch + ')) OR ' +
-              'lower(immutable_unaccent("video"."name")) LIKE lower(immutable_unaccent(' + escapedLikeSearch + '))' +
+              'lower(immutable_unaccent("video"."name")) LIKE lower(immutable_unaccent(' + escapedLikeSearch + ')) OR ' +
+              'lower(immutable_unaccent("video"."description")) % lower(immutable_unaccent(' + escapedSearch + ')) OR ' +
+              'lower(immutable_unaccent("video"."description")) LIKE lower(immutable_unaccent(' + escapedLikeSearch + ')) ' +
```
Comment (Chocobozzz) [inline server/models/video/video.ts]: I don't think a like on description is a good idea

File: server/models/video/video.ts [hunk 3]
```diff
@@ -1082,8 +1082,11 @@ export class VideoModel extends Model<VideoModel> {
               '(' +
               'SELECT "video"."id" FROM "video" ' +
               'WHERE ' +
+              'lower(immutable_unaccent("video"."uuid")) % lower(immutable_unaccent(' + escapedSearch + ')) OR ' +
               'lower(immutable_unaccent("video"."name")) % lower(immutable_unaccent(' + escapedSearch + ')) OR ' +
-              'lower(immutable_unaccent("video"."name")) LIKE lower(immutable_unaccent(' + escapedLikeSearch + '))' +
+              'lower(immutable_unaccent("video"."name")) LIKE lower(immutable_unaccent(' + escapedLikeSearch + ')) OR ' +
+              'lower(immutable_unaccent("video"."description")) % lower(immutable_unaccent(' + escapedSearch + ')) OR ' +
```
Comment (Chocobozzz) [inline server/models/video/video.ts]: We need to create an index of the x first description words/characters first

File: server/models/video/video.ts (hunk 1 above)
Comment (Nutomic) [reply server/models/video/video.ts]: It might be useful if you don't want to type the entire uuid string.

Comment (Chocobozzz) [reply server/models/video/video.ts]: I don't think it would be useful, as we use random uuid v4 so a substring/similar uuid has no meaning

**Comment by Chocobozzz** [comment]
Closing due to inactivity, feel free to request the reopening

**Pull request closed without merging**